arch/arm/core/swap.S

```diff
@@ -324,6 +327,19 @@ _oops:
     blx _do_kernel_oops
     pop {pc}
 
+_escalate_priv:
+    /* check to see if we are already privileged */
+    mrs r2, CONTROL
+    tst r2, #0
```

Comment:
Should be:
`tst r2, #1`
`beq _oops`

`tst r2, #0` the Z flag will always be 1.

Another question is, should we need oops for this situation?

Reply:
on x86 it is currently not a fatal error to invoke a system call in supervisor mode. it will work its just not a great way to go about using an API.

I don't think we should add any logic or checking to catch this. this is a *very* hot code path. it will not happen in practice unless the user is doing something really weird.